cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/TimePartitionedFileSetDatasetAvroSink.java

```diff
@@ -40,43 +43,64 @@
 import org.apache.hadoop.mapreduce.Job;
 
 import java.util.Map;
+import javax.annotation.Nullable;
 
 /**
  * A {@link BatchSink} to write Avro record to {@link TimePartitionedFileSet}
  */
+@Plugin(type = "sink")
+@Name("TimePartitionedFileSetDatasetAvroSink")
+@Description("AVRO Sink with Time Partitioned File Dataset")
 public class TimePartitionedFileSetDatasetAvroSink extends
   BatchSink<GenericRecord, AvroKey<GenericRecord>, NullWritable> {
```

Comment:
Didn't we wanted a AvroSink in the first place. If that has change we can but I think that should go in a different PR completely.